### ISSUE TITLE
Develop the workspace info component

### ISSUE THREAD

**Issue by vanshdvn2505** [opencodeiiita/SkillSync-Frontend#31]
# Workspace Page Info Component  

## **Overview**  
The Workspace Page Info Component is a crucial part of the Workspace Details Page, designed to display comprehensive information about a workspace. This component takes up 50% of the screen (adjustable based on responsiveness) and organizes information into tabbed panes for better accessibility. It highlights key details, including workspace members, shared resources, and scheduled meetings.  

Mentors have additional controls to manage participants, such as removing members. Regular users can only leave the workspace but cannot remove other members.  

---

### **1. Workspace Overview Section:**  
- Displays essential details at the top of the component, including:  
  - **Workspace Name:** The name of the workspace.  
  - **Mentor's Name:** The creator/mentor of the workspace.  
  - **Description:** A brief summary of the workspace’s purpose.  
  - **Type:** Shows whether the workspace is **Public** or **Private** with a clear label or badge.  

### **2. Ongoing Meeting Highlight:**  
- A dedicated area at the top to display information about the currently ongoing meeting (if any):  
  - **Meeting Title:** Name of the ongoing meeting.  
  - **Start Time:** When the meeting started.  
  - **Join Meeting Button:** A clickable button to join the ongoing meeting directly.  
- If no meeting is ongoing, this section is hidden or displays "No Ongoing Meetings."  

### **3. Tabbed Panes for Additional Information:**  
Organize workspace details into tabs:  
- **Members Tab:**  
  - Displays a list of workspace members, including their names, roles (e.g., mentor or participant), and profile pictures (if available).  
  - **Mentor-specific controls:**  
    - Mentors can remove any participant from the workspace.  
  - **Regular User functionality:**  
    - Users can only leave the workspace but cannot remove other participants.  
- **Shared Resources Tab:**  
  - Lists all the resources shared in the workspace, such as files, links, and other materials.  
  - Displays file name, type, and upload date.  
  - Download or open shared resources with a single click.  
- **Scheduled Meetings Tab:**  
  - Shows a list of upcoming meetings with details like:  
    - Meeting Title  
    - Scheduled Date and Time  
    - Action Button to **Join** or **View Details**  

### **4. Responsiveness:**  
- On larger screens, the Workspace Info Component takes up 50% of the Workspace Details Page.  
- On smaller screens, it adjusts to occupy the full width as required.  

---

## **Props**  

| Prop Name               | Type                  | Description                                            |  
|-------------------------|-----------------------|--------------------------------------------------------|  
| `workspaceName`         | String                | Name of the workspace.                                 |  
| `mentorName`            | String                | Name of the mentor who created the workspace.          |  
| `description`           | String                | A brief description of the workspace.                 |  
| `workspaceType`         | String                | Indicates if the workspace is "Public" or "Private".   |  
| `ongoingMeeting`        | Object                | Information about the ongoing meeting (if any).        |  
| `members`               | Array of Objects      | List of workspace members with names, roles, and profile images. |  
| `sharedResources`       | Array of Objects      | List of resources shared in the workspace.            |  
| `scheduledMeetings`     | Array of Objects      | List of upcoming meetings with their details.          |  
| `onJoinMeeting`         | Function              | Callback function to join an ongoing meeting.          |  
| `onRemoveParticipant`   | Function              | Callback for mentors to remove a participant.          |  
| `onLeaveWorkspace`      | Function              | Callback for users to leave the workspace.             |  

---

## **Combined Workspace Details Page**  

This page combines:  
1. **Workspace Info Component:** Displays workspace details and tabbed panes.  
2. **Chat Component:** Allows users to communicate within the workspace.  

### **Page Layout:**  
- **Left Section (50%):** Workspace Info Component.  
- **Right Section (50%):** Chat Component (only accessible to members of the workspace).  

### **Routing:**  
- Add a route for this page in the application:  
  ```javascript
  import WorkspaceDetailsPage from './pages/WorkspaceDetailsPage';

  const routes = [
    { path: '/workspace/:id', component: WorkspaceDetailsPage },
    // other routes...
  ];


---

## **Styling Guidelines**  
- **Overall Layout:**  
  - Allocate 50% width for larger screens.  
  - Ensure smooth transitions and alignment for different tabs.  
- **Tabbed Interface:**  
  - Highlight the active tab with distinct styling (e.g., bold text or color).  
- **Ongoing Meeting Section:**  
  - Use vibrant colors to highlight the ongoing meeting.  
- **Accessibility:**  
  - Ensure the component is keyboard-navigable and follows accessibility standards.  

---


Time: 72hrs

**Comment by ayush-patel1** [comment]
claim

**Comment by ayush-patel1** [comment]
@vanshdvn2505 i have to implement inside compoennt folder right?

**Comment by vanshdvn2505** [comment]
For the workspace info component - yes. But you also have to create workspace details page containing both the chat and info component.

**Comment by vanshdvn2505** [comment]
@ayush-patel1

Thank you for your efforts in contributing to the project and working on the Workspace Info Page. I’ve reviewed your pull request, and unfortunately, it doesn’t meet the requirements outlined in the issue.

Here’s the feedback for improvement:

Workspace Info Page:

The current layout does not follow the described structure. The chat component and info component should be horizontally aligned, each occupying 50% width of the page. This is essential to maintain the expected layout and responsiveness.
The implementation feels incomplete, with everything arranged vertically, which doesn’t align with the design expectations.
Please refer to the issue again for the exact requirements, including the structure, tabbed panes, mentor-specific actions, and routing integration.

Chat Component:

The chat component in your pull request has been altered. However, this component was already developed in a previous issue. Please revert to the original chat component and integrate it as-is without modifications.
I recommend closing this pull request and creating a new one that strictly adheres to the issue's requirements. If you’re unclear about any part of the description or need guidance, feel free to reach out for clarification before starting the rework.

Thank you for understanding, and I appreciate your effort and dedication to the project.

**Comment by ayush-patel1** [comment]
okay sure !! @vanshdvn2505

**Comment by vanshdvn2505** [comment]
@ayush-patel1

Thank you for your work on the recent pull request. The implementation does not meet the design and functionality requirements outlined in the issue description. Additionally, the overall quality and attention to detail in the work are not up to the standard expected for this project.
After reviewing it, I’ve decided to reassign this task to another contributor to ensure the requirements and design standards are fully met.

Please do not continue work on this issue for now. Your effort is appreciated, and I encourage you to explore other issues in the project where you can contribute effectively. If you need guidance on selecting a new issue, feel free to reach out.

Thank you for your understanding, and I look forward to your contributions to the project.

**Comment by official-ashutosh** [comment]
claim

**Comment by Snehalgupta-07** [comment]
claim

**Comment by vanshdvn2505** [comment]
@official-ashutosh Updates ??

**Comment by official-ashutosh** [comment]
@vanshdvn2505 I'll make pr by today

**Comment by vanshdvn2505** [comment]
@official-ashutosh Due to some technical error, I am reassigning this issue to. Please create a new PR for this issue.
